packages/contracts-bedrock/test/governance/ProposalValidator.t.sol

```diff
@@ -578,11 +656,497 @@ contract ProposalValidator_HashProposalWithModule_Test is ProposalValidator_Init
+    }
+
+    function test_submitFundingProposal_governanceFund_succeeds() public {
+        // Calculate expected proposal hash
+        bytes memory votingModuleData =
+            _constructVotingModuleData(optionsDescriptions, optionsRecipients, optionsAmounts, criteriaValue);
+        bytes32 expectedHash = validator.hashProposalWithModule(
+            makeAddr("approvalVotingModule"), votingModuleData, keccak256(bytes(description))
+        );
+
+        // Mock proposalSnapshot to return 0 (proposal doesn't exist in governor)
+        _mockAndExpect(
+            address(governor),
+            abi.encodeCall(IOptimismGovernor.proposalSnapshot, (uint256(expectedHash))),
+            abi.encode(0)
+        );
+
+        // Expect ProposalSubmitted event
+        vm.expectEmit(address(validator));
+        emit ProposalSubmitted(expectedHash, rando, description, ProposalValidator.ProposalType.GovernanceFund);
+
+        // Expect ProposalVotingModuleData event
+        vm.expectEmit(address(validator));
+        emit ProposalVotingModuleData(expectedHash, votingModuleData);
+
+        vm.prank(rando);
+        bytes32 proposalHash = validator.submitFundingProposal(
+            criteriaValue,
+            optionsDescriptions,
+            optionsRecipients,
+            optionsAmounts,
+            description,
+            ProposalValidator.ProposalType.GovernanceFund
+        );
+
+        assertEq(proposalHash, expectedHash);
+    }
+
+    function test_submitFundingProposal_councilBudget_succeeds() public {
+        // Calculate expected proposal hash
+        bytes memory votingModuleData =
+            _constructVotingModuleData(optionsDescriptions, optionsRecipients, optionsAmounts, criteriaValue);
+        bytes32 expectedHash = validator.hashProposalWithModule(
+            makeAddr("approvalVotingModule"), votingModuleData, keccak256(bytes(description))
+        );
+
+        // Mock proposalSnapshot to return 0 (proposal doesn't exist in governor)
+        _mockAndExpect(
+            address(governor),
+            abi.encodeCall(IOptimismGovernor.proposalSnapshot, (uint256(expectedHash))),
+            abi.encode(0)
+        );
+
+        // Expect ProposalSubmitted event
+        vm.expectEmit(address(validator));
+        emit ProposalSubmitted(expectedHash, rando, description, ProposalValidator.ProposalType.CouncilBudget);
+
+        // Expect ProposalVotingModuleData event
+        vm.expectEmit(address(validator));
+        emit ProposalVotingModuleData(expectedHash, votingModuleData);
+
+        vm.prank(rando);
+        bytes32 proposalHash = validator.submitFundingProposal(
+            criteriaValue,
+            optionsDescriptions,
+            optionsRecipients,
+            optionsAmounts,
+            description,
+            ProposalValidator.ProposalType.CouncilBudget
+        );
+
+        assertEq(proposalHash, expectedHash);
+    }
+
+    function test_submitFundingProposal_singleOption_succeeds() public {
```

Comment:
some of the first tests where called twice for both proposal types, the rest of the tests only cover the `Governance Fund` proposals only. I think we could fuzz between the 2 proposal types instead of having 2 times the same test or only test 1 for 1 of the types

Reply:
Same with criteriaValue, we could accept a range of accepted values for criteria value instead of using a single hardcoded one

Reply:
testing edge cases like `test_submitFundingProposal_maximumThreshold_succeeds` is good if you dont fuzz the specific input since it affects the edge case, but the rest of the values that makes sense to be fuzzed should be fuzzed in their acceptable range of the happy path